compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs

```diff
@@ -1209,6 +1209,15 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
             c @ ObligationCauseCode::WhereClauseInExpr(_, _, hir_id, _)
                 if self.tcx.hir().span(*hir_id).lo() == span.lo() =>
             {
+                if let hir::Node::Expr(expr) = self.tcx.parent_hir_node(*hir_id)
+                    && let hir::ExprKind::Call(base, _) = expr.kind
+                    && let hir::ExprKind::Path(hir::QPath::TypeRelative(ty, _)) = base.kind
+                    && ty.span == span
```

Comment:
I'm not following the logic: where in this condition are you checking that the expression is already borrowed?

Reply:
I can add the check for the existing borrow one level up, but I also wanted to avoid suggesting `&str::from("")` when we have `str::from("")` as that won't help either.

Reply:
I see, so the comment below should be saying something like "don't suggest borrowing when this wouldn't fix the problem (because the type is specified by a path instead of inferred)"?

Reply:
Ah I understand my confusion: I had understood "Do not suggest borrowing when we already do so" as "Do not suggest borrowing when the expression is already borrowed"